src/components/form/connection-string-input.jsx

```diff
@@ -58,26 +61,57 @@ class ConnectionStringInput extends React.PureComponent {
    * @returns {String} customUrl.
    */
   getCustomUrl() {
-    return this.props.customUrl;
+    if (this.props.isURIEditable) {
+      return this.props.customUrl;
+    }
+
+    const re = /^(?:(?<prefix>[^:\/?#\s]+):\/{2})?(?:(?<auth>[^@\/?#\s]+)@)?(?<db>[^\/?#\s]+)?(?:\/)?(?<query>.*)$/;
+    const matchObj = re.exec(this.props.customUrl);
+    const { prefix = '', auth, db, query = '' } = matchObj.groups;
+    let safeAuth = '';
+
+    if (auth && auth.includes(':')) {
+      safeAuth = `${auth.split(':')[0]}:*****@`;
+    } else if (auth) {
+      safeAuth = `${auth}@`;
+    }
+
+    return `${prefix}://${safeAuth}${db}/${query}`;
```

Comment:
Just wondering, where's this regex from? Trying it out it looks like it covers all uri cases, but we might want to add tests if it's new.

Reply:
This is a new regex I did.
I was thinking about what is the best way to hide a password. One way is to replace a password with stars in the model, and send two URIs to frontend (one unprotected for the edit mode and one with stars for read-only mode). But in this case, we would build 2 connection objects with 2 driverUrl (protected and unprotected), what kind of messy. And I decided to handle it on the frontend side. We don't apply the regex to user input. We always apply it to driverUrl that is built by our connectin model and has valid syntax, therefore we always can tell that we hide credentials but not something else. Maybe I am missing something, pls let me know if you can think about some edge cases.

Reply:
But you are completely right we need at least some tests for it!

Reply:
You know what :) I have changed my mind. I have found another place where it can be done. So I didn't want to do it in the reflux model of the `compass-connect` and I get your concern about regex, therefore instead I can do it in the `connection-model` itself. It already builds `driverUrl`, I will do a bit of refactoring and it won't take much code to build one more property - `safeUrl`. Then on the frontend side, we simply can decide which of them to use in particular cases.

Reply:
I could see that being useful for the vscode extension if we run into a situation where we want to show the full connection string as well. Sorry for the extra work but I think it'll be well worth it in the long run.

Reply:
No worries! I really like to share ideas and make things better :)

Reply:
Replaces regex with safeUrl from `connection-model`.